packages/rust-bindings/src/aux_functions.rs

```diff
@@ -837,3 +838,100 @@ pub fn serialize_credential_deployment_payload_aux(
 
     Ok(acc_cred_ser)
 }
+
+#[derive(SerdeSerialize, SerdeDeserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct UnsignedCredentialInput {
+    ip_info:                  IpInfo<constants::IpPairing>,
+    global_context:           GlobalContext<constants::ArCurve>,
+    ars_infos:                BTreeMap<ArIdentity, ArInfo<constants::ArCurve>>,
+    id_object: IdentityObjectV1<constants::IpPairing, constants::ArCurve, AttributeKind>,
+    id_cred_sec:              PedersenValue<ArCurve>,
+    prf_key:                  prf::SecretKey<ArCurve>,
+    sig_retrievel_randomness: String,
+    credential_public_keys:   CredentialPublicKeys,
+    attribute_randomness:     BTreeMap<AttributeTag, PedersenRandomness<ArCurve>>,
+    revealed_attributes:      Vec<AttributeTag>,
+    cred_number:              u8,
+}
+
+struct AttributeRandomness(BTreeMap<AttributeTag, PedersenRandomness<ArCurve>>);
+
+#[derive(Debug, Error)]
+pub enum AttributeError {
+    #[error("Missing randomness for given attribute tag.")]
+    NotFound,
+}
+
+impl HasAttributeRandomness<ArCurve> for AttributeRandomness {
+    type ErrorType = AttributeError;
+
+    fn get_attribute_commitment_randomness(
+        &self,
+        attribute_tag: AttributeTag,
+    ) -> Result<PedersenRandomness<ArCurve>, Self::ErrorType> {
+        match self.0.get(&attribute_tag) {
+            Some(v) => Ok(v.clone()),
+            None => Err(AttributeError::NotFound),
+        }
+    }
+}
+
+pub fn create_unsigned_credential_v1_aux(input: UnsignedCredentialInput) -> Result<String> {
+    let chi = CredentialHolderInfo::<constants::ArCurve> {
+        id_cred: IdCredentials {
+            id_cred_sec: input.id_cred_sec,
+        },
+    };
+    let aci = AccCredentialInfo {
+        cred_holder_info: chi,
+        prf_key:          input.prf_key,
+    };
+
+    let blinding_randomness: Value<constants::ArCurve> = concordium_base::common::from_bytes(
+        &mut hex::decode(&input.sig_retrievel_randomness)?.as_slice(),
+    )?;
+    let id_use_data = IdObjectUseData {
+        aci,
+        randomness:
+            concordium_base::id::ps_sig::SigRetrievalRandomness::<constants::IpPairing>::new(
+                *blinding_randomness,
+            ),
+    };
+
+    let context = IpContext::new(&input.ip_info, &input.ars_infos, &input.global_context);
+
+    // And a policy.
+    let mut policy_vec = std::collections::BTreeMap::new();
```

Comment:
Because that is the field name in the struct we are building.
(I have isolated the creation of the Policy into a helper function, because it was actually duplicated)

Reply:
I see, weird that the field's name is `policy_vec` when the type isn't `vec`, but oh well